h/views/api_auth.py

```diff
@@ -31,6 +32,43 @@ def __init__(self, context, request):
     @view_config(request_method='GET',
                  renderer='h:templates/oauth/authorize.html.jinja2')
```

Comment:
Currently IIUC an unsupported `response_mode` value will result in the default redirect behaviour. We could be stricter and check that `response_mode` is one of `None` or `query` here (nb. if we ever add support for implicit grants that would require allowing `fragment` as well I think).

It's not that important IMO so I leave it up to you.